cibuildwheel/docker_container.py

```diff
@@ -57,39 +80,49 @@ def __enter__(self) -> "DockerContainer":
             network_args = ["--network=host"]
 
         shell_args = ["linux32", "/bin/bash"] if self.simulate_32_bit else ["/bin/bash"]
+
         subprocess.run(
             [
-                "docker",
+                self.container_engine,
                 "create",
                 "--env=CIBUILDWHEEL",
                 f"--name={self.name}",
                 "--interactive",
-                "--volume=/:/host",  # ignored on CircleCI
                 *network_args,
-                *cwd_args,
+                # Z-flags is for SELinux
+                "--volume=/:/host:Z",  # ignored on CircleCI
                 self.docker_image,
                 *shell_args,
             ],
+            env=self.env,
             check=True,
         )
+
         self.process = subprocess.Popen(
             [
-                "docker",
+                self.container_engine,
                 "start",
                 "--attach",
                 "--interactive",
                 self.name,
             ],
             stdin=subprocess.PIPE,
             stdout=subprocess.PIPE,
+            env=self.env,
         )
 
         assert self.process.stdin and self.process.stdout
         self.bash_stdin = self.process.stdin
         self.bash_stdout = self.process.stdout
 
         # run a noop command to block until the container is responding
-        self.call(["/bin/true"])
+        self.call(["/bin/true"], cwd="")
```

Comment:
The use of empty string as semantically different from None is a little too tricky. I think there's probably a better way to do this, that handles the working dir stuff at the start, rather than setting for each command.

Reply:
I'm fine with it as long as it works.

Reply:
Maybe we can just do `cwd="/"` for these first couple commands instead. That would be preferable to me.

Reply:
I just removed the cwd argument, which reverts this to its original state.

Reply:
Oh cool, that works too :)

Reply:
I just noticed that leaving the cwd argument out means that there's output like this in the terminal:

```text
    + /bin/true
/bin/bash: line 2: cd: /project: No such file or directory
    + mkdir -p /project
/bin/bash: line 8: cd: /project: No such file or directory
```

Which are inconsequential errors, but the user wouldn't know that.

So maybe `cwd="/"` would be better after all.

Reply:
Seems reasonable. I pushed up that change.